test-helpers/src/connection/kafka/mod.rs

```diff
@@ -164,6 +165,28 @@ impl KafkaConsumer {
             }
         }
     }
+
+    /// The offset to be committed should be lastProcessedMessageOffset + 1.
+    pub fn assert_commit_offsets(&self, offsets: HashMap<TopicPartition, i64>) {
+        match self {
+            #[cfg(feature = "kafka-cpp-driver-tests")]
+            Self::Cpp(cpp) => cpp.commit(&offsets, rdkafka::consumer::CommitMode::Sync),
+            Self::Java(java) => java.commit_sync(&offsets),
+        }
+
+        let partitions = offsets.keys().cloned().collect::<HashSet<_>>();
+
+        let responses = match self {
+            #[cfg(feature = "kafka-cpp-driver-tests")]
+            Self::Cpp(cpp) => cpp.committed_offsets(partitions),
+            Self::Java(java) => java.committed(partitions),
+        };
+
+        for (topic_partition, offset) in offsets {
```

Comment:
Since the kafka driver will return a variable number of elements in responses, it would be good to make an assertion on the number of responses to make sure extra responses were not returned.

Would it be appropriate to assert that `assert_eq!(offsets.len(), responses.len())`?

Reply:
yeah will add this assertion